tools/analysis_tools/get_flops.py

```diff
@@ -43,46 +50,128 @@ def parse_args():
     return args
 
 
-def main():
-    args = parse_args()
+def inference(args, logger):
+    logger.warning(
+        'if you want test all configs flops, please make sure torch>=1.12')
+    # In torch<1.12, torch.jit don't support some configs compute flops
+    # like configs/detectors, configs/yolact, so if you want to count the
+    # flops mentioned above, you need update torch version. other configs
+    # can compute flops in torch<1.12
+
+    config_name = Path(args.config)
+    if not config_name.exists():
+        logger.error(f'{config_name} not found.')
+
+    cfg = Config.fromfile(args.config)
+    cfg.work_dir = tempfile.TemporaryDirectory().name
+    cfg.log_level = 'WARN'
+    if args.cfg_options is not None:
+        cfg.merge_from_dict(args.cfg_options)
+
+    init_default_scope(cfg.get('default_scope', 'mmdet'))
+
+    # TODO: The following usage is temporary and not safe
+    # use hard code to convert mmSyncBN to SyncBN. This is a known
+    # bug in mmengine, mmSyncBN requires a distributed environment，
+    # this question involves models like configs/strong_baselines
+    if hasattr(cfg, 'head_norm_cfg'):
+        cfg['head_norm_cfg'] = dict(type='SyncBN', requires_grad=True)
+        cfg['model']['roi_head']['bbox_head']['norm_cfg'] = dict(
+            type='SyncBN', requires_grad=True)
+        cfg['model']['roi_head']['mask_head']['norm_cfg'] = dict(
+            type='SyncBN', requires_grad=True)
 
     if len(args.shape) == 1:
         h = w = args.shape[0]
     elif len(args.shape) == 2:
         h, w = args.shape
     else:
         raise ValueError('invalid input shape')
-    ori_shape = (3, h, w)
     divisor = args.size_divisor
```

Comment:
Why is this parameter needed? Shouldn't it be read directly from the configuration?